src/model.js

```diff
@@ -2586,6 +2586,8 @@ class Model {
       options
     );
 
+    instance.isNewRecord = false;
```

Comment:
Although here `instance === result`, we should change naming here. May be rename `result` to `record` and then set `record.isNewRecord = false`

Reply:
Result is actually `[instance, boolean]`. Should I do this?
```suggestion
    const record = [result];
    record.isNewRecord = false;
```

Reply:
Yes, that works

Reply:
Looks like tests are failing with this commit